### DISCUSSION TITLE
Group 4 Discussion

### DISCUSSION THREAD

**Discussion by leninoc** [Digital-Preservation-Coalition/PythonStudyGroups#18] [General]
Discussion for Group 4 - here is our work Google Document https://docs.google.com/document/d/12yKhYUZsQOj9xnb0aD5KD4pXqRtDWxlX-Vcgd4en89w/edit?usp=sharing

**Comment by leninoc** [comment]
TODO list from our first call 

- Choose one online tutorial and put notes into this shared doc
- Set up environment for Python
- Jan to look into different installer/enviroments

Google doc now updated with list of tutorials and very basic HOWTO install Python
Please have a look at the tutorials and share which one you would like to follow

**Reply by jeni-kim** [reply]
Thanks, Jan! I was thinking Library Carpentry since you had previously used that one, but since it's under construction, it may not be the best option? I will continue with the _Bite-Size Python_ book, but I'm thinking maybe the W3Schools tutorial? I'm open to what everyone else wants to do, though.

**Reply by stape1** [reply]
Hi and thanks Jan.  [W3Schools](https://www.w3schools.com/python/) python tutorial looks good to me.

**Reply by AIbbots1** [reply]
Thanks for setting this up, Jan :)
I'm happy to go with W3Schools. It look simple enough that I may even understand it 😬

**Reply by leninoc** [reply]
hi, thanks for your feedback. Looking at it and then on our list of options (posting below for your convenience) I am starting to think that maybe we dont need to go with one tutorial but we can do two. For example the Library Carpentry Intro for Libraries (Nr. 3 below) and W3C (Nr. 9), they actually do complement each other I think. I also like Nr. 5. Michelle Janowieczki’s Python lessons.
Hopefully more ppl will post their ideas here before next week.

PS - I checked the Library carpentry and its complete, so we can use it. 
cheers
Jan

1. https://infx511.github.io/ “Introduction to programming” course book shared my one of my colleagues also in this group NC
2. [Bite Size Python](https://www.amazon.com/Bite-Size-Python-Introduction-Programming/dp/1119643813) (book)
3. [Python Intro for Libraries: Summary and Setup (librarycarpentry.org)](https://librarycarpentry.org/lc-python-intro/) 
4. [Get Started Tutorial for Python in Visual Studio Code](https://code.visualstudio.com/docs/python/python-tutorial)
5. Michelle Janowieczki’s Python lessons for librarians [Home · Wiki · Michelle Janowiecki / Python lessons for librarians · GitLab](https://gitlab.com/mjanowiecki/python-lessons-for-librarians/-/wikis/home) (7 lessons)
6. David Beazley Practical Python Programming [Practical Python Programming | practical-python (dabeaz-course.github.io)](https://dabeaz-course.github.io/practical-python/Notes/Contents.html)
7. [The Python Tutorial — Python 3.12.1 documentation](https://docs.python.org/3/tutorial/)
8. [Python Programming Tutorial for Beginners - After Hours Programming](https://www.afterhoursprogramming.com/tutorial/python/python-overview/)
9. [Introduction to Python (w3schools.com)](https://www.w3schools.com/python/python_intro.asp)
10. [Automate the Boring Stuff with Python](https://automatetheboringstuff.com/) book and tutorial

**Comment by leninoc** [comment]
hi everyone, 
I think we wanted to agree on a tutorial we are going to use by February 14th. From the feedback above I think its the Library carpentry and W3C tutorials who are the "winners". 
Our next call is on 27/28 February, so I guess I will personally aim to start with both. 
- W3C - will try to go through the lessons up to "python strings"
- Library carpentry - first 4 lessons/parts.
Also will go back to Mimo https://mimo.org which i have used years ago on my phone, time to revisit!
Take care everyone,
Jan

**Reply by jeni-kim** [reply]
This sounds good to me! I will begin working through those. Thanks for coordinating, Jan!

**Comment by leninoc** [comment]
hi everyone,
it was great to see you yesterday on our 2nd call. Minutes are [here](https://docs.google.com/document/d/12yKhYUZsQOj9xnb0aD5KD4pXqRtDWxlX-Vcgd4en89w/edit?usp=sharing), thanks to minute takers!
I think for next call in March we agreed to continue with our own learning, plus we also thought we should give it a go and try the ["delete unwanted file" scenario/practice](https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/blob/main/How-Tos/Use%20Case%20How%20To%20-%20Deleting%20Unwated%20FIles.md).
For that I promised to prepare a folder with few random files which we can use. I have copied few files from [OPF test file corpus](https://github.com/openpreserve/format-corpus/tree/master). Then I thought we might be using 2 different scenarios: 
1) simple - where all files are in one folder and we are trying to delete say all **.sam files**; and 
2) different scenario where we have 2 subfolders and we are also trying to delete **.sam files** from one of the subfolders. But its up to you which scenario you choose. 

Also if you have a real usecase feel free to share with the group.

Here are the files zipped
**Scenario 1** - simple - [python_test_files_simple.zip](https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/files/14440099/python_test_files_simple.zip)
MD5 checksum 8cdedd7fd8c5ecb347351d1916fb158e *python_test_files_simple.zip

**Scenario 2** - subfolders - [python_test_files_subfolders.zip](https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/files/14440101/python_test_files_subfolders.zip)
MD5 checksum 8741edb7267bcbd0d51475b166c6240a *python_test_files_subfolders.zip

Good luck! And remember, the discussion is here for us to discuss, share and ask.
cheers,
Jan

**Reply by jeni-kim** [reply]
This is awesome! Thanks, Jan! I’ll get to work and I’ll be back here if I have any questions.

**Reply by jeni-kim** [reply]
I did run into an issue (how to pull all the files that end with .sam), but wanted to try to solve it by myself. Anyhow, this is late in the game, but I just thought I'd post this here in case it's helpful to anyone: https://stackoverflow.com/questions/3964681/find-all-files-in-a-directory-with-extension-txt-in-python

**Reply by leninoc** [reply]
hi Jeni, same here, that same issue, i guess my task was too ambitious, but i plan to try to find the solution myself, lets see if its going to be the same with yours!

**Reply by leninoc** [reply]
i ended up using file.endswith(), not the glob.glob

**Comment by AIbbots1** [comment]
Hi everyone,
I have too many competing priorities to dedicate the time this pursuit deserves right now. Our digital archive goes live to agencies in 8 short weeks and between writing procedures, advice for agencies and staff/ agency training there is not much time for anything else. Good luck with it all, I'll check in on this page every so often to see how you're all getting on :) Amy

**Reply by jeni-kim** [reply]
We'll miss you, Amy, but best of luck with your digital archive launch! That sounds amazing (albeit tiring)!

**Comment by critchleyn** [comment]
Hi all,

Due to a competing priority, I am going to have to miss today's meeting.
I look forward to reviewing the notes!

Best,
Nicole

**Reply by jeni-kim** [reply]
Oh no, I'm so sorry! I totally missed this today!! 🤦🏻‍♀️ I did complete the deleting files exercise and it was both exhilarating and terrifying at how easy Python made it. Anyhow, if anyone took notes from today or if the rest of you decided on a new exercise, please let us know.

**Reply by leninoc** [reply]
Hi Jeni and all,
we were wondering what happened, sorry I should have texted you or something. It was just myself and Sally in the end. We went through the task we got and discussed few more things. I took minutes, they are available in our usual document. 
For the next task we thought it might be fun/useful to go through the "batch renaming" scenario [here](https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/blob/main/How-Tos/Use%20Case%20-%20How%20to%20batch%20rename%20files.md), even if it means just to follow up the explanation on the website. I have found out that even following up excercise like this and trying to understand how each step is done gives me a lot. And maybe on next call we can each present what we have done?
If anyone has another scenario you would like to try please let us know.
cheers,
Jan

**Comment by leninoc** [comment]
hi everyone,
I wont be able to join the call later today, really sorry for that. Looking forward to seeing you all next time!
Jan
